Units/parser-javascript.r/js-es6-mixin.d/input.js

```diff
@@ -0,0 +1,28 @@
+
+class Foo {
+  foo() { return "foo"; }
+}
+
+class Bar extends Foo {
+  bar() { return "bar"; }
+}
+
+function Mixin(superclass) {
+  return class extends superclass {
+    hello() {
+      return "hello from mixin";
+    }
+  }
+}
+
```

Comment:
Yeah got me a bit of reading to understand it -- especially when the examples also make use of arrow functions, another thing I'm not familiar with.  But the actual concept is fairly simple when you accepted the dynamic nature of it: a function that returns an anonymous class inheriting from a parameter.  It has the strange property of not being an "instance" of the mixin as the `Mixin` function returns a new class each time, but apart from that it's fairly "simple".